src/compiler/transformers/automatic-key-insertion/index.ts

```diff
@@ -91,16 +91,10 @@ export const performAutomaticKeyInsertion = (transformCtx: ts.TransformationCont
    * @returns the result of handling the node
    */
   function jsxElementVisitor(node: ts.Node): ts.VisitResult<ts.Node> {
-    if (ts.isJsxExpression(node)) {
-      // we don't have the static analysis chops to dive into a JSX expression
-      // (arbitrary JS code delimited by curly braces in JSX) in order to
-      // determine whether it's safe to add keys to JSX nodes within it, so we
-      // bail here instead.
-      return node;
-    } else if (ts.isConditionalExpression(node)) {
-      // we're going to encounter the same problem here that we encounter with
-      // multiple return statements, so we don't try to transform the arms of
-      // the conditional.
+    if (ts.isCallExpression(node) || ts.isPropertyAccessExpression(node) || ts.isConditionalExpression(node)) {
+      // we're going to encounter the same problems here that we encounter with
```

Comment:
Since we've change this conditional to remove `isJsxExpression` and add `isCallExpression`+`isPropertyAccessExpression`, I think it might make sense to update this comment. I remember you doing this original work, but I think some of the context of this comment is a little lost on me now (TBH, I don't remember what the multiple return statement problem is 😆).

Reply:
yeah, I think that actually the `isPropertyAccessExpression` check isn't necessary overall, I was thinking of situations like

```ts
<div>{ foo.bar(<div />) }</div>
```

where we might not want to transform the `<div />` because we don't know what will happen to it (`.bar` could do anything!) but actually in this case the whole expression `foo.bar(<div />)` is a `CallExpression` where the callee is a `PropertyAccessExpression`. So just checking for `isPropertyAccessExpression` would only guard against situations like

```ts
<div>{ foo.bar }</div>
```

where the property access expression can't have any children that are JSX nodes anyway...

so anyway yeah I think I will remove it and then fix up the comments

Reply:
I updated the code here and also fleshed out a comment in the `findRenderMethodVisitor` helper function to explain a bit more the 'multiple return' problem, lmk what you think!